charts/ark-solr/templates/statefulset.yaml

```diff
@@ -0,0 +1,63 @@
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: {{ include "solr.fullname" . }}-statefulset
+  labels:
+    {{- include "solr-helm-charts.labels" . | nindent 6 }}
+  namespace: {{ .Release.Namespace | quote }}
+spec:
+  replicas: 1
+  selector: 
+    matchLabels:  
+      {{- include "solr-helm-charts.labels" . | nindent 6 }}
+  serviceName: {{ include "solr.fullname" . }}-headless
+  template:
+    metadata:
+      name: ark-solr
+      labels:
+        {{- include "solr-helm-charts.labels" . | nindent 8 }}
+    spec:
+      initContainers:
+      containers:
+      - name: solr
+        image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+        ports:
+        - containerPort: 8983
+          name: solr-port
+          protocol: TCP
+        command:
+        livenessProbe:
+        readinessProbe: 
+        resources:
+        volumeMounts:
+        - name: solr-data
+          mountPath: /data
+        - name: solr-conf
+          mountPath: /opt/solr/server/solr/configsets/_default/conf
+      volumes:
+        - name: solr-conf
+          configMap: 
+            name: {{ include "solr.configmap-name" . }}
+        {{- if and .Values.persistence.enabled .Values.persistence.existingClaim }}
+        - name: solr-data
+          persistentVolumeClaim:
+            claimName: {{ printf "%s" (tpl .Values.persistence.existingClaim .) }}
+        {{- else if not .Values.persistence.enabled }}
```

Comment:
@naveen-armedia Is it really useful to plan not to have persistence for Solr?

Reply:
@fabricetriboix I believe this is how we can standardize the persistence volume across all our applications.

Reply:
@naveen-armedia I don't really understand what you wrote... Could you please elaborate on this, for example by stating the exact mechanism to standardize persistent volumes across our Helm charts?